main.go

```diff
@@ -94,6 +103,62 @@ func showErrorMessages(action string, errorMessages map[string]error) error {
 	return nil
 }
 
+type ConfProfile struct {
+	RemoteConfiguration bool `json:"active"`
+	Compliance          bool `json:"compliance"`
+	Insights            bool `json:"insights"`
+	Remediation         bool `json:"remediations"`
+}
+
+// getConfProfile will retrieve the profile the system is configured
+func getConfProfile(client *http.Client) (*ConfProfile, error) {
+	var profile ConfProfile
+
+	url, err := GuessAPIURL()
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := client.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	// Read the response body
+	defer res.Body.Close()
+
+	if res.StatusCode != 200 {
+		return nil, fmt.Errorf("cannot get system profile: %s", res.Status)
+	}
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(body, &profile); err != nil {
+		return nil, fmt.Errorf("cannot marshal HTTP response body: %w", err)
+	}
+
+	return &profile, nil
+}
+
+// showConfProfile shows a list of the system profile enabled features gathered from API
+// https://github.com/RedHatInsights/config-manager/blob/master/internal/http/v2/openapi.json
+func showConfProfile(p *ConfProfile) {
+	if p != nil {
+		s := reflect.ValueOf(p).Elem()
```

Comment:
Wow. This is a whole lot of `reflect` code to just get some field names and true/false values. Maybe marshalling the JSON response into a `map[string]bool` would be better than a struct type.

```go
var profile map[string]bool
if err := json.Unmarshal(body, &profile); err != nil {
    ...
}
```

Then calling `showConfProfile` could just enumerate the map:

```go
for service, status := range profile {
    if status {
        fmt.Printf("%v, ", service)
    }
}
```

Reply:
I have modified this a bit. Instead of printing by service, I added everything as an array of string and uses the `strings` module to separate them by commas and then printed as a title (with the first letter in uppercase). This way the output looks like:

```
● Enabled console.redhat.com services: Remote Configuration, Insights, Remediations, Compliance
```

Instead of:
```
● Enabled console.redhat.com services: remote configuration, insights, remediations, compliance,
```

Reply:
I had to change this a bit because the `strings.Title` was deprecated. I opted to no add more complexity to the function and the output of this lines is:

```
● Enabled console.redhat.com services: remote configuration, insights, remediations, compliance
```